dbt/adapters/athena/impl.py

```diff
@@ -56,6 +58,80 @@ def convert_number_type(cls, agate_table: agate.Table, col_idx: int) -> str:
     def convert_datetime_type(cls, agate_table: agate.Table, col_idx: int) -> str:
         return "timestamp"
 
+    def _parse_lf_tags(self, lf_tags_expr: Optional[str]) -> Dict[str, str]:
+        lf_tags = {}
+        if lf_tags_expr:
+            lf_tags = dict([tag.split("=") for tag in lf_tags_expr.split(",")])
+
+        return lf_tags
+
+    @available
+    def add_lf_tags_to_table(self, database: str, table: str, lf_tags_expr: Optional[str] = None):
+        conn = self.connections.get_thread_connection()
+        session = conn.handle.session
+
+        lf_tags = self._parse_lf_tags(lf_tags_expr)
+        lf_client = session.client("lakeformation")
+
+        if lf_tags:
+            lf_client.add_lf_tags_to_resource(
+                Resource={
+                    "Table": {
+                        "DatabaseName": database,
+                        "Name": table,
+                    },
+                },
+                LFTags=[
+                    {
+                        "TagKey": key,
+                        "TagValues": [
+                            value,
+                        ],
+                    }
+                    for key, value in lf_tags.items()
+                ],
+            )
+
+    @available
+    def add_lf_tags_to_database(self, database: str):
+        conn = self.connections.get_thread_connection()
+        session = conn.handle.session
+
+        lf_tags_expr = conn.credentials.lf_tags
+        lf_tags = self._parse_lf_tags(lf_tags_expr)
+        lf_client = session.client("lakeformation")
+
+        if lf_tags:
+            lf_client.add_lf_tags_to_resource(
+                Resource={
+                    "Database": {"Name": database},
+                },
+                LFTags=[
+                    {
+                        "TagKey": key,
+                        "TagValues": [
+                            value,
+                        ],
+                    }
+                    for key, value in lf_tags.items()
+                ],
+            )
+
+    @available
+    def get_work_group_output_location(self) -> Optional[str]:
+        conn = self.connections.get_thread_connection()
+        creds = conn.credentials
+        session = conn.handle.session
+
+        athena_client = session.client("athena")
```

Comment:
same as here: https://github.com/dbt-athena/dbt-athena/pull/159/files#r1117410404
use the context manager lock